distributed/scheduler.py

```diff
@@ -1447,8 +1450,13 @@ def state(self, value: TaskStateState) -> None:
 
     def add_dependency(self, other: TaskState) -> None:
         """Add another task as a dependency of this task"""
+        if self.dependencies is None:
+            self.dependencies = set()
```

Comment:
As already pointed out above, there are very, very few cases where a Task doesn't walk through this code path. Therefore, I suggest to keep this strictly non-optional

Reply:
https://github.com/dask/distributed/pull/8331/commits/7df524092d3fd185da7d1be750a7be43cbf89430